### Tarea 5.4

Ejercicio sacado de "Basic Probability Theory", R. B. Ash, Dover, 1970.

En una cierta población, el $20\%$ de las personas tiene una enfermedad. Se hace un examen a los miembros de la población que tiene la característica de dar positivo el $90\%$ de las veces si la persona tiene la enfermedad y el $30\%$ de las veces si la persona no tiene la enfermedad (falso positivo). A los que dan positivo se les suministra una droga que puede causar manchas rojas en la piel en un $20\%$ de los pacientes. 

Si una persona tiene esas manchas rojas. ¿Cuál es la probabilidad de que haya tenido la enfermedad?

Para esta solución se utilizará el Teorema de Bayes:

$$
P(A|B) = \frac{P(B|A) P(A)}{P(B)}
$$

Primero se deben de definir los parámetros a medir, en este caso, la cnatidad de personas en la población puede variar al gusto de usuario, esto es así ya que no es tan relevante la cantidad en la población, sino más bien el proceso para calcular que tan probable es que haya un infectado que tenga manchas rojas:

In [1]:
poblacion = 1000000 #Cantidad de habitantes en la población. El usuario puede modificar este valor.
sanos = 0.8*poblacion #Porcentaje personas sanas en la población. Un 80%.
enfermos = 0.2*poblacion #Porcentaje personas enfermas en la población. Un 20%.
positivo = 0.9*enfermos #Porcentaje de enfermos que dan positivo en el exámen. Un 90%.
falso_positivo = 0.3*sanos #Porcentaje de sanos que dan falso positivo. Un 30%.
manchas_rojas_real = 0.2*positivo #Porcentaje de personas con manchas enfermas. Un 20% de los positivos.
manchas_rojas_falso = 0.2*falso_positivo #Porcentaje de personas con manchas sanas. Un 20% de los falsos positivos.
manchas_rojas_total = manchas_rojas_real+manchas_rojas_falso #Porcentaje total de personas con manchas rojas.

Método Bayesiano, donde se considera el porcentaje real de la cantidad de personas con manchas rojas que estan infectadas, en comparación a la cantidad total de personas con manchas rojas:

In [2]:
manchas_rojas_real/(manchas_rojas_total)

0.42857142857142855

Como se puede observar, la probabilidad de que una persona que tenga manchas rojas esté también infectada es $\approx 42.9\%$.